templates/interface/Debian.erb

```diff
@@ -9,7 +9,7 @@ allow-hotplug <%= @interface %>
 <% if @manage_address -%>
     address <%= @manage_address %>
 <% end -%>
-<% if @manage_hwaddr -%>
+<% if @manage_address and ! @manage_address.empty? -%>
```

Comment:
Sorry, I don't follow you.
Why on dhcp interfaces the  hwaddr entry should not be there?
Are you sure that the line:
<% if @manage_address and ! @manage_address.empty? -%>
should be before hwaddr and not address?